sdk/identity/Azure.Identity/src/MsalPublicClient.cs

```diff
@@ -74,25 +74,30 @@ public virtual async ValueTask<AuthenticationResult> AcquireTokenSilentAsync(str
         public virtual async ValueTask<AuthenticationResult> AcquireTokenInteractiveAsync(string[] scopes, string claims, Prompt prompt, bool async, CancellationToken cancellationToken)
         {
             IPublicClientApplication client = await GetClientAsync(async, cancellationToken).ConfigureAwait(false);
-            if (async)
+
+#pragma warning disable AZC0109 // Misuse of 'async' parameter.
+            if (!async && !IdentityCompatSwitches.DisableInteractiveBrowserThreadpoolExecution)
```

Comment:
I don't think this code is equivalent. In the sync case with the switch off, we wouldn't block the task anymore.

Reply:
I'm not sure I follow your comment. If we were called in a sync context and the user didn't set the compat flag we use the `Task.Run` workaround. Otherwise if we were called in an async context, or the compat flag is set we fallback to the old behavior.  That is the intended behavior. Are your saying that this is not the actual behavior, or that my intended behavior is wrong?

Reply:
My bad, I was looking at a partial change.